utils/_context/virtual_machines.py

```diff
@@ -137,10 +138,12 @@ def get_cache_name(self):
         vm_cached_name = f"{self.name}_"
         if self.get_provision().lang_variant_installation:
             vm_cached_name += f"{self.get_provision().lang_variant_installation.id}_"
+        vm_cached_installations = ""
         for installation in self.get_provision().installations:
             if installation.cache:
-                vm_cached_name += f"{installation.id}_"
-        return vm_cached_name
+                vm_cached_installations += f"{installation.id}_"
```

Comment:
Shouldn't we use a hash of the object instead of the id?

Reply:
thinking about hash all the yml content...but for now should work. I added because the character limitation on the ami names

Reply:
When does the id change?

Reply:
Changing the names of the scenarios. My idea (for the near future) is to extract some hash or unique identifier for the content of yml files, if something changes it will be other id